File: benchmark/results/java-invocation-inline/run-001/b-baseline/output/com/metaengine/demo/catalog/services/PricingEngine.java
package com.metaengine.demo.catalog.services;

import com.metaengine.demo.catalog.aggregates.Product;
import java.util.List;

/** Domain service that computes pricing for catalog products. */
public class PricingEngine {
    /** Create a priced product from the provided partial input. */
    public Product create(Product input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find a priced product by id. Returns null when not found. */
    public Product findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List up to {@code limit} priced products. */
    public List<Product> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Remove pricing for the given id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-multilang/run-001/b-baseline/output/metaengine_demo/catalog/services/pricing_engine.py
"""PricingEngine for the catalog domain."""
from typing import Optional

from metaengine_demo.catalog.aggregates.product import Product


class PricingEngine:
    """PricingEngine domain service."""

    def create(self, input: Optional[Product]) -> Product:
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Product]:
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Product]:
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        raise NotImplementedError("not implemented")
